src/main/java/net/neoforged/neoforge/event/level/BlockEvent.java

```diff
@@ -99,6 +99,49 @@ public void setExpToDrop(int exp) {
         }
     }
 
+    /**
+     * Fired when a block is destroyed AFTER drops have been determined, but have not yet spawned.
+     * - Cancellation will result in the drops not being spawned.
+     * - Breaker may be null if the block was not destroyed by an entity.
+     * - The drops list is immutable and cannot be modified. Use LootTableModifiers instead
+     * - It is safe to modify the Block in this event, as it has already been replaced.
+     */
+    public static class DestroyedEvent extends BlockEvent implements ICancellableEvent {
+        private final List<ItemStack> drops;
+        private final Entity destroyingEntity;
+        private final ItemStack tool;
+
+        private boolean dropXpWhenCancelled;
+
+        public DestroyedEvent(LevelAccessor level, BlockPos pos, BlockState state, List<ItemStack> drops, @Nullable Entity destroyer, ItemStack tool) {
+            super(level, pos, state);
+            this.drops = drops;
+            this.destroyingEntity = destroyer;
+            this.tool = tool;
+            this.dropXpWhenCancelled = true;
+        }
+
+        public void setDropXpWhenCancelled(boolean shouldDrop) {
+            this.dropXpWhenCancelled = shouldDrop;
+        }
+
+        public List<ItemStack> getDrops() {
```

Comment:
We should seek to provide a list of `ItemEntity` here instead of a list of `ItemStack`, and make the list mutable (allowing people to add/remove entities).

Unfortunately I think this would require some static ThreadLocal hacks or replicating the logic in `popResource`, let me see what options we have.

Reply:
I have replicated the entity creation and spawn logic to provide a list of ItemEntities instead. Though this doesn't seem like an ideal solution...

Reply:
Pushed a commit implementing the capture drops mechanism.